src/lib/plugins/backdrop/backdrop.service.ts

```diff
@@ -0,0 +1,28 @@
+import { Injectable, Renderer2, ElementRef } from '@angular/core';
+import { Guard } from 'ng2-qgrid/core/infrastructure/guard';
+
+@Injectable()
+export class BackdropService {
+	element: ElementRef;
+	isVisible = true;
+
+	hide() {
+		Guard.notNull(this.element, 'element');
+		if (this.isVisible) {
+			this.element.nativeElement.classList.add('q-grid-backdrop-inactive');
+			this.isVisible = false;
+		}
+	}
+
+	reveal() {
+		Guard.notNull(this.element, 'element');
+		if (!this.isVisible) {
+			this.element.nativeElement.classList.remove('q-grid-backdrop-inactive');
+			this.isVisible = true;
+		}
+	}
+
+	reset() {
```

Comment:
could we directly do element = null and git rid of reset , because it's became not dual you set element using one way and unset with another